modules/identity-provider-configuring-apache-request-header.adoc

```diff
@@ -1,114 +1,89 @@
+using the request header identity provider.
+
+.Prerequisites
+
+* Obtain the `mod_auth_gssapi` module from the
 link:https://access.redhat.com/solutions/392003[Optional channel].
-Install the following packages:
+You must have following packages installed on your local machine:
++
+** `httpd`
+** `mod_ssl`
+** `mod_session`
+** `apr-util-openssl`
+** `mod_auth_gssapi`
+
+* Generate a CA for validating requests that submit the trusted header. Use the
+file containing this CA certificate to create a ConfigMap; specify the ConfigMap
+in the `oauth/cluster` object's `ca` field.
+
+* Generate a client certificate for the proxy. You can generate this certificate
+by using any x509 certificate tooling. The client certificate must be signed by
+the CA you generated for validating requests that submit the trusted header.
+
+While this reference configuration uses Apache's `mod_auth_gssapi`, it is not
+required; other proxies can easily be used if the following requirements are
+met:
+
+. Block the `X-Remote-User` header from client requests to prevent spoofing.
+. Enforce client certificate authentication in the `RequestHeaderIdentityProvider`
+configuration.
+. Require the `X-Csrf-Token` header be set for all authentication requests using
+the challenge flow.
+. Only the `/oauth/authorize` endpoint and its subpaths should be proxied, and
+redirects should not be rewritten to allow the backend server to send the client
+to the correct location.
+. The URL that proxies to `\https://<namespace_route>/oauth/authorize` must end
+with `/authorize` (with no trailing slash). For example:
 +
 ----
-# yum install -y httpd mod_ssl mod_session apr-util-openssl mod_auth_gssapi
+https://proxy.example.com/login-proxy/authorize?...
 ----
-
-. Generate a CA for validating requests that submit the trusted header. Use this CA
-as the file name for `clientCA` in the identity provider configuration.
++
+should proxy to
 +
 ----
-# oc adm ca create-signer-cert \
-  --cert='/etc/origin/master/proxyca.crt' \
-  --key='/etc/origin/master/proxyca.key' \
-  --name='openshift-proxy-signer@1432232228' \
-  --serial='/etc/origin/master/proxyca.serial.txt'
+https://<namespace_route>/oauth/authorize?...
 ----
 +
-[NOTE]
-====
-The `oc adm ca create-signer-cert` command generates a certificate that is valid
-for five years. This can be altered with the `--expire-days` option, but for
-security reasons, it is recommended to not make it greater than this
-value.
-
-// Run `oc adm` commands only from the first master listed in the Ansible host inventory file,
-// by default *_/etc/ansible/hosts_*.
-====
-
-. Generate a client certificate for the proxy. You can generate this certificate
-by using any x509 certificate tooling, including the `oc adm` CLI:
+. Subpaths of the URL that proxies to `\https://<namespace_route>/oauth/authorize`
+must proxy to subpaths of `\https://<namespace_route>/oauth/authorize`. For
+example:
```

Comment:
@stlaz this one repeats the same URL twice in the step instruction (i.e., `https://<namespace_route>/oauth/authorize`).  The example, however, converts the first URL differently, not exactly matching the initial description:

```https://proxy.example.com/login-proxy/authorize/approve?...```
&rarr;
```https://<namespace_route>/oauth/authorize/approve?...```

This was also done similarly in 3.11 (except with `<master>`).

Would this make more sense?
> Subpaths of the URL that proxies to `https://proxy.example.com/login-proxy/authorize` must proxy to subpaths of `https://<namespace_route>/oauth/authorize`.

Reply:
No, `https://proxy.example.com/login-proxy/authorize` is the proxy address, nothing proxies to it